naga/src/front/wgsl/parse/mod.rs

```diff
@@ -2612,4 +2628,47 @@ impl Parser {
         }
         Ok(brace_nesting_level + 1)
     }
+
+    fn diagnostic_filter<'a>(
+        &self,
+        lexer: &mut Lexer<'a>,
+    ) -> Result<Option<DiagnosticFilter>, Error<'a>> {
+        lexer.expect(Token::Paren('('))?;
+
+        let (severity_control_name, severity_control_name_span) = lexer.next_ident_with_span()?;
+        let new_severity = Severity::from_ident(severity_control_name).ok_or(
+            Error::DiagnosticInvalidSeverity {
+                severity_control_name_span,
+            },
+        )?;
+
+        lexer.expect(Token::Separator(','))?;
+
+        let (diagnostic_name_token, diagnostic_name_token_span) = lexer.next_ident_with_span()?;
+        let diagnostic_rule_name = if lexer.skip(Token::Separator('.')) {
+            // Don't try to validate these name tokens on two tokens, which is conventionally used
+            // for third-party tooling.
+            lexer.next_ident_with_span()?;
+            None
+        } else {
+            Some(diagnostic_name_token)
+        };
+        let diagnostic_rule_name_span = diagnostic_name_token_span;
+
+        let filter = diagnostic_rule_name
+            .map(|name| {
+                FilterableTriggeringRule::from_ident(name).ok_or(Error::DiagnosticInvalidRuleName {
+                    diagnostic_rule_name_span,
+                })
```

Comment:
> Using an unrecognized triggering rule consisting of a single diagnostic name-token should trigger a warning from the user agent.

The spec says this should be a warning even for the single `diagnostic name-token` path, but we currently have no way of accumulating warnings. Maybe we should ignore it for now?

Reply:
I've filed follow-up work for this:

- https://github.com/gfx-rs/wgpu/issues/5295
- https://github.com/gfx-rs/wgpu/issues/6458

Reply:
I'll bring in the reporting functionality I wrote for `Severity`, which should do a `log::warn!(…)`.

Reply:
Latest force-push as of 4d609af00156588e352c5afa5c1fd8d755c27c9b should resolve this; it uses `Severity` to determine how to proceed with either logging an error, or returning an error to break control flow.

Reply:
@teoxoy: Will let you `Resolve conversation` once you determine that the changed code does not offend your sensibilities. 🙂